pkg/controller/master/backup/backup_metadata.go

```diff
@@ -132,18 +133,40 @@ func (h *MetadataHandler) syncVMBackup(target *settings.BackupTarget) error {
 		return err
 	}
 
+	var namespaceFolders []string
+	var requiredMovingFilePaths []string
 	for _, fileName := range fileNames {
-		backupMetadata, err := loadBackupMetadataInBackupTarget(filepath.Join(metadataFolderPath, fileName), bsDriver)
+		filePath := filepath.Join(metadataFolderPath, fileName)
+		if bsDriver.FileExists(filePath) {
+			requiredMovingFilePaths = append(requiredMovingFilePaths, filePath)
+		} else {
+			namespaceFolders = append(namespaceFolders, filePath)
+		}
+	}
+
+	// collect all vmbackup metadata file paths in vmbackupMetadataFilePaths
+	vmbackupMetadataFilePaths := make([]string, len(requiredMovingFilePaths))
+	copy(vmbackupMetadataFilePaths, requiredMovingFilePaths)
+	for _, namespaceFolder := range namespaceFolders {
+		fileNames, err := bsDriver.List(filepath.Join(metadataFolderPath))
```

Comment:
This appears to be identical to the call on line 131. I don't think the original value from 131 has been mutated, so could it be used here instead of calling bsDriver.List inside the loop?

Reply:
Sorry, I used a wrong file path here. I would like to like files under namespace folder. Thanks for catching this!